### DISCUSSION TITLE
Merge statement

### DISCUSSION THREAD

**Discussion by Pipboyguy** [duckdb/duckdb#4601] [Q&A]
Is there a `MERGE INTO` statement available in the API as yet?

**Comment by Alex-Monahan** [comment]
Not yet, but here is a workaround!
#61

**Comment by davetapley** [comment]
Not sure it's what @Pipboyguy was looking for, but I have a use case where I'd like to merge two entirely separate DBs, which don't _necessarily_ have the same columns.

I've got a proof of concept implementation where they at least have common tables.
It introspects using `information_schema.columns`, and does `INSERT OR REPLACE INTO ... BY NAME`.
I'm sure it's missing lots of edge cases 😁 

This makes it a bit awkward though:
- https://github.com/duckdb/duckdb/discussions/4639

---

I'd be interested if others have this use case?

**Comment by strongbad03** [comment]
Folks please upvote with 👍 or it doesn't appear higher in GitHub sort 🙏

**Comment by carlopi** [comment]
Somewhat relevant: [Add support for MERGE INTO](https://github.com/duckdb/duckdb/pull/18135#top), an in-flight PR to duckdb/duckdb

**Reply by Tishj** [reply]
Only *somewhat* ;)

**Comment by hannes** [comment]
This was implemented
